### ISSUE TITLE
Add support for integration tests

### ISSUE THREAD

**Issue by nepalez** [evilmartians/evil-client#18]
Quoting a comment of @palkan :

What about adding some support for integration tests too? I mean something like that:

```ruby
before { stub_operation(CatClient, "cats.fetch") }

# or
before { stub_operation(CatClient, /cats\..*/) }

# or with return value
before { stub_operation(CatClient, /.*/).and_return(double("Cat")) }

# or with constraints
before { stub_operation(CatClient, "cats.fetch").with(token: "valid") }

# etc
```

Now I'm testing evil-client integration the following way:

```ruby
let(:candidates) do 
  double('candidates').tap do |stub|
    allow(client).to receive(:candidates).and_return(stub)
  end
end

it "works" do
  expect(candidates).to receive(:create).with(
        ...
   ).and_return(double('response', id: 'checkr-123'))

  # check some code that handles the response
end
```

**Comment by nepalez** [comment]
@palkan Could you look at
https://github.com/evilmartians/evil-client/blob/1.1.0/spec/unit/rspec/stub_evil_client_operation_spec.rb
https://github.com/evilmartians/evil-client/blob/1.1.0/spec/unit/rspec/expect_evil_client_operation_spec.rb

**Comment by palkan** [comment]
👍 LGTM

One comment: what about make shorter names and remove "evil", 'cause users will use it to test `Some::Client` – nothing about `evil-client` itself.
So, just: `stub_client_operation` and `expect_client_operation` – these names look descriptive and unique enough (I've tried to search on GitHub and found nothing).

**Comment by nepalez** [comment]
Well, at first I were thinking of `stub_operation` etc., then added `evil_client` just to prevent collisions.

You're right, `_client_` should be enouth

**Comment by nepalez** [comment]
Thank you! This will be included to `v1.1.0`
